mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -480,66 +480,83 @@ def create_from_app_path(app_path):
     @staticmethod
     def create_text_preparation_pipeline(
         language: str = ENGLISH_LANGUAGE_CODE,
-        preprocessors: List[str] = None,
+        preprocessors: Tuple[Union[str, Preprocessor]] = None,
         regex_norm_rules: List[Dict] = None,
         keep_special_chars: str = None,
-        normalizers: List[str] = None,
-        tokenizer: Tokenizer = None,
-        stemmer: Stemmer = None,
+        normalizers: Tuple[Union[str, Normalizer]] = None,
+        tokenizer: Union[str, Tokenizer] = None,
+        stemmer: Union[str, Stemmer] = None,
     ):
         """Static method to create a TextPreparationPipeline instance.
 
         Args:
             language (str, optional): Language as specified using a 639-1/2 code.
-            preprocessors (List[str]): List of preprocessor class names.
-            regex_norm_rules (List[Dict]): List of regex normalization rules represented as
-                dictionaries. ({"pattern":<pattern>, "replacement":<replacement>})
-            normalizers (List[str]): List of normalizer class names.
-            tokenizer (str): Class name of Tokenizer to use.
-            stemmer (str): Class name of Stemmer to use.
+            preprocessors (Tuple[Union[str, Preprocessor]]): List of preprocessor class
+                names.
+            regex_norm_rules (List[Dict]): List of regex normalization rules represented
+                as dictionaries. ({"pattern":<pattern>, "replacement":<replacement>})
+            normalizers (Tuple[Union[str, Preprocessor]]): List of normalizer class names.
+            tokenizer (Union[str, Tokenizer]): Class name of Tokenizer to use.
+            stemmer (Union[str, Stemmer]): Class name of Stemmer to use.
 
         Returns:
             TextPreparationPipeline: A TextPreparationPipeline class.
         """
-        preprocessors = (
-            [PreprocessorFactory.get_preprocessor(p) for p in preprocessors]
+
+        # Instantiate Preprocessors
+        instantiated_preprocessors = (
+            TextPreparationPipelineFactory._instantiate_pipeline_components(
+                Preprocessor, preprocessors, language
+            )
             if preprocessors
             else [NoOpPreprocessor()]
         )
 
+        # Update Regex Normalization Exception Characters as Specified in the Config
         if keep_special_chars:
-            RegexNormalizerRuleFactory.EXCEPTION_CHARS = keep_special_chars
+            RegexNormalizerRuleFactory.set_exception_chars(keep_special_chars)
```

Comment:
Since this is a class var, I dont think we need to have getters and setters. My previous suggestion was if we made it into an instance var.

Reply:
Got it, removed getters and setters